paasta_tools/cli/cmds/validate.py

```diff
@@ -601,91 +720,80 @@ def validate_autoscaling_configs(service_path):
-                    config = get_config_file_dict(
-                        os.path.join(
-                            soa_dir,
-                            service,
-                            f"{instance_config.get_instance_type()}-{cluster}.yaml",
-                        ),
-                        use_ruamel=True,
-                    )
-                    if config[instance].get("cpus") is None:
-                        # cpu autoscaled, but using autotuned values - can skip
-                        continue
 
-                    cpu_comment = _get_comments_for_key(
-                        data=config[instance], key="cpus"
-                    )
-                    # we could probably have a separate error message if there's a comment that doesn't match
-                    # the ack pattern, but that seems like overkill - especially for something that could cause
-                    # a DAR if people aren't being careful.
-                    if (
-                        cpu_comment is None
-                        or re.search(
-                            pattern=OVERRIDE_CPU_AUTOTUNE_ACK_PATTERN,
-                            string=cpu_comment,
+                try:
+                    if autoscaling_params["metrics_provider"] == "active-requests":
+                        _validate_active_requests_autoscaling_configs(
+                            instance_config, autoscaling_params
+                        )
+
+                    elif autoscaling_params["metrics_provider"] in {"uwsgi", "piscina"}:
+                        _validate_uwsgi_piscina_autoscaling_configs(autoscaling_params)
+
+                    elif autoscaling_params["metrics_provider"] == "arbitrary_promql":
+                        _validate_arbitrary_promql_autoscaling_configs(
+                            autoscaling_params
                         )
-                        is None
+
+                    elif (
+                        autoscaling_params["metrics_provider"] == "cpu"
+                        # to enable kew autoscaling we just set a decision policy of "bespoke", but
+                        # the metrics_provider is (confusingly) left as "cpu"
+                        and autoscaling_params.get("decision_policy") != "bespoke"
+                        and not should_skip_cpu_override_validation
                     ):
-                        returncode = False
-                        print(
-                            failure(
-                                msg=f"CPU override detected for a CPU-autoscaled instance in {cluster}: {service}.{instance}. Please read "
-                                "the following link for next steps:",
-                                link="y/override-cpu-autotune",
+                        _validate_cpu_autoscaling_configs(autoscaling_params)
+                        # Do some extra validation below: we don't abstract that into the above function
```

Comment:
i really should have tossed all this nonsense into a function when i initially wrote this :p